src/specialized.jl

```diff
@@ -97,7 +97,7 @@ function parse_string(ps::MemoryParserState, b)
             c = ps.utf8data[s]
             if c == LATIN_U  # Unicode escape
                 ps.s = s + 1
-                for c in string(read_unicode_escape!(ps)).data
+                for c in Vector{UInt8}(string(read_unicode_escape!(ps)))
```

Comment:
could use the new `codeunit` function here to avoid allocating a `Vector{UInt8}` object.  Or we could re-implement [the Char-encoding logic](https://github.com/JuliaLang/julia/blob/887815c69d4511bb16f8b72a1a80bc57f97e044d/base/strings/string.jl#L186-L204) in order to avoid allocating a `String` at all.